compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -454,26 +452,24 @@ pub(super) fn response_no_constraints<'tcx>(
     goal: Canonical<'tcx, impl Sized>,
     certainty: Certainty,
 ) -> QueryResult<'tcx> {
-    let var_values = goal
-        .variables
-        .iter()
-        .enumerate()
-        .map(|(i, info)| match info.kind {
-            CanonicalVarKind::Ty(_) | CanonicalVarKind::PlaceholderTy(_) => {
-                tcx.mk_ty(ty::Bound(ty::INNERMOST, ty::BoundVar::from_usize(i).into())).into()
-            }
-            CanonicalVarKind::Region(_) | CanonicalVarKind::PlaceholderRegion(_) => {
-                let br = ty::BoundRegion {
-                    var: ty::BoundVar::from_usize(i),
-                    kind: ty::BrAnon(i as u32, None),
-                };
-                tcx.mk_region(ty::ReLateBound(ty::INNERMOST, br)).into()
+    let var_values =
+        tcx.mk_substs(goal.variables.iter().enumerate().map(|(i, info)| -> ty::GenericArg<'tcx> {
+            match info.kind {
+                CanonicalVarKind::Ty(_) | CanonicalVarKind::PlaceholderTy(_) => {
+                    tcx.mk_ty(ty::Bound(ty::INNERMOST, ty::BoundVar::from_usize(i).into())).into()
+                }
+                CanonicalVarKind::Region(_) | CanonicalVarKind::PlaceholderRegion(_) => {
+                    let br = ty::BoundRegion {
+                        var: ty::BoundVar::from_usize(i),
+                        kind: ty::BrAnon(i as u32, None),
+                    };
+                    tcx.mk_region(ty::ReLateBound(ty::INNERMOST, br)).into()
+                }
+                CanonicalVarKind::Const(_, ty) | CanonicalVarKind::PlaceholderConst(_, ty) => tcx
+                    .mk_const(ty::ConstKind::Bound(ty::INNERMOST, ty::BoundVar::from_usize(i)), ty)
+                    .into(),
             }
-            CanonicalVarKind::Const(_, ty) | CanonicalVarKind::PlaceholderConst(_, ty) => tcx
-                .mk_const(ty::ConstKind::Bound(ty::INNERMOST, ty::BoundVar::from_usize(i)), ty)
-                .into(),
-        })
-        .collect();
+        }));
```

Comment:
:thinking: this should just be `CanonicalVarValues::make_identity`? so we should replace it

Reply:
But we don't have a CanonicalVarValues here, we have a CanonicalVarInfos, so we can't call make_identity?

Reply:
Actually, there are no call-sites for `make_identity` anymore :thinking: 

I'll just make it take `CanonicalVarInfos`